GNUmakefile

```diff
@@ -85,7 +85,7 @@ pkg/linux_amd64/nomad: $(SOURCE_FILES) ## Build Nomad for linux/amd64
 
 pkg/linux_arm/nomad: $(SOURCE_FILES) ## Build Nomad for linux/arm
 	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=linux GOARCH=arm CC=arm-linux-gnueabihf-gcc-5 \
+	@CGO_ENABLED=1 GOOS=linux GOARCH=arm GOARM=6 CC=arm-linux-gnueabihf-gcc-5 \
```

Comment:
Shouldn't this be overridable via an env variable so people can build with `GOARM=6` or `GOARM=7`?

Reply:
According to the [GoArm page](https://github.com/golang/go/wiki/GoArm), The go compiler defaults to the current chipset. The GOARM environment variable has to be manually set, and would normally return a blank value. Is there a way for the GNUMake file use a default if the environment variable isn't set, defaulting to the chipset value?
The GNUMake docs make note of using variable overrides, but I'm not too familiar with GNUMake.

Reply:
Does ARMv6 work on ARMv7? If so defaulting to ARMv6 seems like the right thing to do for our 32bit ARM release.

Reply:
Not my area of expertise either, but I think a statically linked binary would work but not a dynamically linked one. Doesn't our build process do the latter though?

Reply:
Yes, we use CGO so it's dynamically linked.

Reply:
The build flag set to GOARM=6 will produce a binary which works on both Pi 3 and Pi Zero, I have tested this on both hardwares by building this branch on a Raspberry Pi 3.  I also removed the CC flag to use the default gcc.

However, the linked binary in the conversation would not run on my pi.

Interestingly the default setting for Go on my Pi is GOARM=6 however this is an arm 7

```bash
$ go env
GOARCH="arm"
GOBIN="/home/pi/go/bin"
GOEXE=""
GOHOSTARCH="arm"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/pi/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_arm"
GCCGO="gccgo"
GOARM="6"
CC="gcc"
GOGCCFLAGS="-fPIC -marm -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build581517738=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```